cli/app.go

```diff
@@ -425,32 +425,114 @@ var app = &cli.App{
+									UsageText: fmt.Sprintf("viam data dataset add-by-id <%s> <%s> <%s> <%s> [other options]",
+										datasetFlagDatasetID, dataFlagOrgID, dataFlagLocationID, dataFlagFileIDs),
+									Flags: []cli.Flag{
+										&cli.StringFlag{
+											Name:     datasetFlagDatasetID,
+											Usage:    "dataset ID to which data will be added",
+											Required: true,
+										},
+										&cli.StringFlag{
+											Name:     dataFlagOrgID,
+											Usage:    "org ID to which data belongs",
+											Required: true,
+										},
+										&cli.StringFlag{
+											Name:     dataFlagLocationID,
+											Usage:    "location ID to which data belongs",
+											Required: true,
+										},
+										&cli.StringSliceFlag{
+											Name:     dataFlagFileIDs,
+											Usage:    "file IDs of data belonging to specified org and location",
+											Required: true,
+										},
+									},
+									Action: DataAddToDataset,
 								},
-								&cli.StringFlag{
-									Name:     dataFlagLocationID,
-									Usage:    "location ID to which data belongs",
-									Required: true,
-								},
-								&cli.StringSliceFlag{
-									Name:     dataFlagFileIDs,
-									Usage:    "file IDs of data belonging to specified org and location",
-									Required: true,
+
+								{
+									Name:      "filter",
+									UsageText: fmt.Sprintln("viam data dataset add filter [other options]"),
+									Flags: []cli.Flag{
+										&cli.StringFlag{
+											Name:     datasetFlagDatasetID,
+											Usage:    "dataset ID to which data will be added",
+											Required: true,
+										},
+										&cli.StringSliceFlag{
+											Name:  dataFlagOrgIDs,
+											Usage: "orgs filter",
+										},
+										&cli.StringSliceFlag{
+											Name:  dataFlagLocationIDs,
+											Usage: "locations filter",
+										},
+										&cli.StringFlag{
+											Name:  dataFlagRobotID,
+											Usage: "robot-id filter",
+										},
+										&cli.StringFlag{
+											Name:  dataFlagPartID,
+											Usage: "part id filter",
+										},
+										&cli.StringFlag{
+											Name:  dataFlagRobotName,
+											Usage: "robot name filter",
+										},
+										&cli.StringFlag{
+											Name:  dataFlagPartName,
+											Usage: "part name filter",
+										},
+										&cli.StringFlag{
+											Name:  dataFlagComponentType,
+											Usage: "component type filter",
+										},
+										&cli.StringFlag{
+											Name:  dataFlagComponentName,
+											Usage: "component name filter",
+										},
+										&cli.StringFlag{
+											Name:  dataFlagMethod,
+											Usage: "method filter",
+										},
+										&cli.StringSliceFlag{
+											Name:  dataFlagMimeTypes,
+											Usage: "mime types filter",
+										},
+										&cli.UintFlag{
+											Name:  dataFlagParallelDownloads,
+											Usage: "number of download requests to make in parallel",
```

Comment:
This might be left if you refactor the below into one common method, but technically not downloading anything here (vs in `binaryData`); just processing in parallel.

Reply:
I just omitted here, since we should are just fixing the parallel processing # in the backend for adding filter data to dataset. I think this is fine.